src/CMakeLists.txt

```diff
@@ -0,0 +1,357 @@
+# Copyright (c) 2019 LG Electronics, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set(IS_WEBOS FALSE)
+set(IS_AGL FALSE)
+
+string(TOLOWER ${PLATFORM_NAME} PLATFORM)
+
+if(${PLATFORM} STREQUAL "webos")
+    set(IS_WEBOS TRUE)
+elseif(${PLATFORM} MATCHES ".*agl")
+    set(IS_AGL TRUE)
+else()
+    message(FATAL_ERROR "Unsupported platform: ${PLATFORM}")
+endif()
+
+if(EXISTS ${CHROMIUM_SRC_DIR})
+    set(CHROMIUM_INCLUDE_DIRS ${CHROMIUM_SRC_DIR})
+    set(CHROMIUM_LDFLAGS -lcbe)
+else()
+    message(FATAL_ERROR "Invalid Chromium source dir: ${CHROMIUM_SRC_DIR}")
+endif()
+
+# Disable compiler-specific extensions
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+set(WAM_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+set(WAM_DEST_DIR ${CMAKE_INSTALL_PREFIX})
+set(WAM_PKGCONFIG_DIR "${CMAKE_SOURCE_DIR}/files/pkgconfig")
+set(WAM_COMMON_DEFINES PRELOADMANAGER_ENABLED _REENTRANT)
+
+message(STATUS "** Build Type: ${CMAKE_BUILD_TYPE}")
+message(STATUS "** Build platform: ${PLATFORM}")
+message(STATUS "** Chromium src dir: ${CHROMIUM_SRC_DIR}")
+message(STATUS "** Destination dir: ${WAM_DEST_DIR}")
+
+find_package(Boost COMPONENTS filesystem REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_search_module(GLIB REQUIRED glib-2.0)
+pkg_search_module(LIBXML REQUIRED libxml-2.0)
+pkg_search_module(JSONCPP REQUIRED jsoncpp)
+
+if(IS_WEBOS)
+    set(CMAKE_AUTOMOC ON)
+    if(USE_LUNA_SERVICE)
+        pkg_search_module(LIBLUNAPREFS REQUIRED luna-prefs)
```

Comment:
(Can be fixed later, when adding support for building with cmake in webOS): this is missing luna-service2 dependency, that is something different from luna-prefs. It can be detected with pkg-config luna-service2